src/webui/src/components/PackageSidebar/index.jsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import API from '../../../utils/api';
+
+import classes from './style.scss';
+import getRegistryURL from '../../../utils/getRegistryURL';
+
+export default class PackageSidebar extends React.Component {
+  state = {
+    lastUpdate: 'Loading',
+    recentReleases: [],
+    author: null,
+    contributors: null,
+    showAllContributors: false,
+    dependencies: {}
+  };
+
+  constructor(props) {
+    super(props);
+    this.showAllContributors = this.showAllContributors.bind(this);
+  }
+
+  async componentDidMount() {
+    await this.loadPackageData(this.props.packageName);
+  }
+
+  async componentWillReceiveProps(newProps) {
+    if (newProps.packageName !== this.props.packageName) {
+      await this.loadPackageData(newProps.packageName);
+    }
+  }
+
+  async loadPackageData(packageName) {
+    let packageMeta;
+
+    try {
+      packageMeta = (await API.get(`sidebar/${packageName}`)).data;
+    } catch (err) {
+      this.setState({
+        failed: true
+      });
+      return;
+    }
+
+    let lastUpdate = 0;
+    Object.keys(packageMeta._uplinks).forEach((upLinkName) => {
+      const status = packageMeta._uplinks[upLinkName];
+
+      if (status.fetched > lastUpdate) {
+        lastUpdate = status.fetched;
+      }
+    });
+
+    let recentReleases = Object.keys(packageMeta.time).map((version) => {
+      return {
+        version,
+        time: packageMeta.time[version]
+      };
+    });
+    recentReleases = recentReleases.slice(recentReleases.length - 3, recentReleases.length).reverse();
+
+    this.setState({
+      lastUpdate: lastUpdate ? (new Date(lastUpdate)).toLocaleString() : '',
+      recentReleases,
+      author: packageMeta.latest.author,
+      contributors: packageMeta.latest.contributors,
+      dependencies: packageMeta.latest.dependencies,
+      showAllContributors: _.size(packageMeta.latest.contributors) <= 5
+    });
+  }
+
+  showAllContributors() {
+    this.setState({
+      showAllContributors: true
+    });
+  }
+
+  render() {
+    let {author, contributors, recentReleases, lastUpdate, showAllContributors, dependencies} = this.state;
+
+    let uniqueContributors = [];
```

Comment:
there can be `uniqueContributors` function to get unique contributors. It'll be easy to test.

Reply:
👍